### ISSUE TITLE
Add configuration to no-inferrable-types to ignore annotated function parameters

### ISSUE THREAD

**Issue by Arnavion** [palantir/tslint#1095]
The `no-inferrable-types` currently complains if a function parameter has both a type annotation and a default value.
#### TypeScript code being linted

``` ts
function foo(bar: number = 0) { }
```

with `tslint.json`:

``` json
{
    "rules: {
        "no-inferrable-types": true
    }
}
```
#### Actual behavior

`LHS type (number) inferred by RHS expression, remove type annotation`

---

It would be nice if the rule had a configuration to skip function parameters. I prefer documenting the types of function parameters explicitly regardless of whether they're default-initialized or not.

**Comment by rrogowski** [comment]
I believe this was addressed by https://github.com/palantir/tslint/pull/1190.

Running `tslint` on the above code example with the following rules gives no lint error.

```json
{
  "rules": {
    "no-inferrable-types": [ true, "ignore-params" ]
  }
}

```

**Comment by JoshuaKGoldberg** [comment]
Oh yes, that's correct! Whoohoo!
